### ISSUE TITLE
ClientConnection.peer_certificates() to return Some() regardless of CertificateError::UnknownIssuer

### ISSUE THREAD

**Issue by nworbnhoj** [rustls/rustls#1819]
-  [x]  I've searched the issue tracker for similar requests

### Problem

I would like to inspect the Certificate chain offered by a host that I do not (yet) trust.

Currently,  `peer_certificates()` will return  `None` if the `ClientConnection` experiences a ` CertificateError::UnknownIssuer`

### Request

I am requesting that `peer_certificates()` will return `Some()` **regardless** of failure via `CertificateError::UnknownIssuer`

### Context

Rather than beginning with a fully populated `RootCertStore` (ie `webpki-roots`), the concept is to begin with an empty `RootCertStore` and have the opportunity to inspect and consider the Certificate chain offered by a host **before** adding trusted Certificates to the `RootCertStore`.

In this scenario, it is important that the Certificate chain is valid in all respects, but for ` CertificateError::UnknownIssuer`. 

This scenario extends to self-signed Certificates.

### Alternatives

I have a functional work-around via a custom `ServerCertVerifier` but it introduces unwelcome complexity:
* this obviously necessitates calling `ClientConfig.dangerous()`
* the custom `ServerCertVerifier` returns `Ok(ServerCertVerified::assertion())` on `CertificateError::UnknownIssuer`
*  a bogus Certificate is added to the `RootCertStore` passed to `WebPkiServerVerifier::builder()` to avoid tripping the early "sanity check" for an empty `RootCertStore`. https://github.com/rustls/rustls/blob/cf098b08d32b8f9636faf3dd82a1b8d4a85a146e/rustls/src/webpki/server_verifier.rs#L365-L374
* The private `rustls::crypto::ring::SUPPORTED_SIG_ALGS` is vendored in

**Comment by djc** [comment]
I don't think we we'll want to support this (which IMO is off the beaten/secure by default path) inside rustls. I suppose we can probably make `SUPPORTED_SIG_ALGS` public?

**Comment by ctz** [comment]
> I suppose we can probably make `SUPPORTED_SIG_ALGS` public?

This is actually already available publicly, it is: `rustls::crypto::ring::default_provider().signature_verification_algorithms`

**Comment by nworbnhoj** [comment]
Thank you for your consideration and clarification.

I certainly subscribe to the "secure by default" path.

I did not envisage any change to the behaviour or gravity of `CertificateError::UnknownIssuer` - but only to expose the integral, but untrusted, Certificate chain. I imagined that `peer_certificates()` is purely informational - while the role of `CertificateError` is definitive and actionable. I was suprised when `peer_certificates()` returned `None`.

Thank you for the pointer to `rustls::crypto::ring::default_provider().signature_verification_algorithms` - I had missed that.

I will persist with `ClientConfig.dangerous()` and a custom `ServerCertVerifier`

Thanks you for all of your great work with `rustls`.
